modules/userId/index.js

```diff
@@ -465,13 +470,112 @@ function getUserIdsAsEids() {
 }
 
+/**
+* This function will be exposed in the global-name-space so that userIds can be refreshed after initialization.
+* @param {RefreshUserIdsOptions} options
+*/
+function refreshUserIds(options, callback) {
+  let submoduleNames = options.submoduleNames;
+  if (!submoduleNames) {
+    submoduleNames = [];
+  }
+
+  initializeSubmodulesAndExecuteCallbacks(function() {
+    let consentData = gdprDataHandler.getConsentData()
+
+    const storedConsentData = getStoredConsentData();
+    setStoredConsentData(consentData);
+
+    // gdpr consent with purpose one is required, otherwise exit immediately
+    let {userIdModules, hasValidated} = validateGdprEnforcement(submodules, consentData);
+    if (!hasValidated && !hasGDPRConsent(consentData)) {
+      utils.logWarn(`${MODULE_NAME} - gdpr permission not valid for local storage or cookies, exit module`);
+      return;
+    }
+
+    let callbackSubmodules = [];
+    for (let submodule of userIdModules) {
+      if (submoduleNames.length > 0 &&
+        submoduleNames.indexOf(submodule.submodule.name) === -1) {
+        continue;
+      }
+
+      utils.logInfo(`${MODULE_NAME} - refreshing ${submodule.submodule.name}`);
+      populateSubmoduleId(submodule, consentData, storedConsentData, true);
+
+      if (utils.isFn(submodule.callback)) {
+        callbackSubmodules.push(submodule);
+      }
+    }
+
+    if (callbackSubmodules.length > 0) {
+      processSubmoduleCallbacks(callbackSubmodules);
+    }
+
+    if (callback) {
+      callback();
+    }
+  });
+}
+
 /**
  * This hook returns updated list of submodules which are allowed to do get user id based on TCF 2 enforcement rules configured
  */
 export const validateGdprEnforcement = hook('sync', function (submodules, consentData) {
   return {userIdModules: submodules, hasValidated: consentData && consentData.hasValidated};
 }, 'validateGdprEnforcement');
 
+function populateSubmoduleId(submodule, consentData, storedConsentData, forceRefresh) {
+  // There are two submodule configuration types to handle: storage or value
+  // 1. storage: retrieve user id data from cookie/html storage or with the submodule's getId method
+  // 2. value: pass directly to bids
+  if (submodule.config.storage) {
+    let storedId = getStoredValue(submodule.config.storage);
+    let response;
+
+    let refreshNeeded = false;
+    if (typeof submodule.config.storage.refreshInSeconds === 'number') {
+      const storedDate = new Date(getStoredValue(submodule.config.storage, 'last'));
+      refreshNeeded = storedDate && (Date.now() - storedDate.getTime() > submodule.config.storage.refreshInSeconds * 1000);
+    }
+
+    if (!storedId || refreshNeeded || !!forceRefresh || !storedConsentDataMatchesConsentData(storedConsentData, consentData)) {
+      // No id previously saved, or a refresh is needed, or consent has changed. Request a new id from the submodule.
+      response = submodule.submodule.getId(submodule.config.params, consentData, storedId);
```

Comment:
what do folks think about passing `forceRefresh` into this function so if a submodule wants to treat a refresh differently for some reason, they can?

Reply:
@icflournoy do you have thoughts on this?

Reply:
i've created a new issue to track interest in this: https://github.com/prebid/Prebid.js/issues/5818